src/Atmos/Model/bc_energy.jl

```diff
@@ -203,3 +199,90 @@ function atmos_energy_normal_boundary_flux_second_order!(
     # NOTE: difference from design docs since normal points outwards
     fluxᵀn.energy.ρe -= C_h * ρ_avg * normu_int⁻_tan * (MSE - MSE_int)
 end
+
+"""
+    NishizawaEnergyFlux(fn) :: EnergyBC
+Calculate the net inward energy flux across the boundary following Nishizawa and Kitamura (2018).
+Return the flux (in W m^-2).
+"""
+struct NishizawaEnergyFlux{FNTM, FNX} <: EnergyBC
+    fn_z0::FNX
+    fn_T_and_q_tot::FNTM
+end
+function atmos_energy_boundary_state!(
+    nf,
+    bc_energy::NishizawaEnergyFlux,
+    atmos,
+    args...,
+) end
+function atmos_energy_normal_boundary_flux_second_order!(
+    nf,
+    bc_energy::NishizawaEnergyFlux,
+    atmos,
+    fluxᵀn,
+    n⁻,
+    state⁻,
+    diffusive⁻,
+    hyperdiffusive⁻,
+    aux⁻,
+    state⁺,
+    diffusive⁺,
+    hyperdiffusive⁺,
+    aux⁺,
+    t,
+    state_int⁻,
+    diffusive_int⁻,
+    aux_int⁻,
+)
+    FT = eltype(state⁻)
+    # Interior state
+    u_int⁻ = state_int⁻.ρu / state_int⁻.ρ
+    u_int⁻_tan = projection_tangential(atmos, aux_int⁻, u_int⁻)
+    normu_int⁻_tan = norm(u_int⁻_tan)
+    q_tot_int =
+        atmos.moisture isa DryModel ? FT(0) :
+        state_int⁻.moisture.ρq_tot / state_int⁻.ρ
+    # recover thermo state
+    ts_int = recover_thermo_state(atmos, state_int⁻, aux_int⁻)
+    x_in =
+        MArray{Tuple{3}, FT}(FT[normu_int⁻_tan, dry_pottemp(ts_int), q_tot_int])
+
+    # Boundary state
+    T, q_tot = bc_energy.fn_T_and_q_tot(atmos, state⁻, aux⁻, t)
+    x_s = MArray{Tuple{3}, FT}(FT[FT(0), T, q_tot])
+
+    ## Initial guesses for MO parameters, these should be a function of state.
+    LMO_init = FT(100) # Initial value so that ξ_init<<1
+    u_star_init = FT(0.1) * normu_int⁻_tan
+    th_star_init = T
+    qt_star_init = q_tot
+    MO_param_guess = MArray{Tuple{4}, FT}(FT[
+        LMO_init,
+        u_star_init,
+        th_star_init,
+        qt_star_init,
+    ])
+
+    # Roughness and interior heights
+    z_0 = bc_energy.fn_z0(atmos, state⁻, aux⁻, t, normu_int⁻_tan)
```

Comment:
A minor point. We may not need `normu_int⁻_tan` in the input?

Reply:
I left the same signature as the `C_drag` coefficients for now, in case we want to use some formulation of the roughness height that depends on the tangential velocity.